providers/base/bin/wifi_nmcli_test.py

```diff
@@ -42,26 +43,82 @@ def print_cmd(cmd):
     print("+", cmd)
 
 
-def cleanup_nm_connections():
-    print_head("Cleaning up NM connections")
-    cmd = "nmcli -t -f TYPE,UUID,NAME c"
+def _get_nm_wireless_connections():
+    cmd = "nmcli -t -f TYPE,UUID,NAME,STATE connection"
     print_cmd(cmd)
-    output = sp.check_output(cmd, shell=True)
+    output = sp.check_output(shlex.split(cmd))
+    connections = {}
     for line in output.decode(sys.stdout.encoding).splitlines():
-        type, uuid, name = line.strip().split(":", 2)
+        type, uuid, name, state = line.strip().split(":", 3)
         if type == "802-11-wireless":
-            print("Deleting connection", name)
-            cmd = "nmcli c delete {}".format(uuid)
+            connections[name] = {"uuid": uuid, "state": state}
+    return connections
+
+
+def get_nm_activate_connection():
+    print_head("Get NM activate connection name")
+    connections = _get_nm_wireless_connections()
+    for name, value in connections.items():
+        state = value["state"]
+        uuid = value["uuid"]
+        if state == "activated":
+            print("Activated Connection: {} {}".format(name, uuid))
+            return uuid
+    return ""
+
+
+def turn_up_connection(uuid):
+    # uuid can also be connection name
+    print_head("Turn up NM connection")
+    cmd = "nmcli c up {}".format(uuid)
+    print("Turn up {}".format(uuid))
+    activate_uuid = get_nm_activate_connection()
+    if uuid == activate_uuid:
+        print("{} state is already activated".format(uuid))
+        return None
+    try:
+        print_cmd(cmd)
+        sp.call(shlex.split(cmd))
+    except Exception as e:
+        print("Can't turn on {}: {}".format(uuid, str(e)))
+
+
+def turn_down_nm_connections():
+    print_head("Turn off NM all connections")
+    connections = _get_nm_wireless_connections()
+    for name, value in connections.items():
+        uuid = value["uuid"]
+        print("Turn down connection", name)
+        try:
+            cmd = "nmcli c down {}".format(uuid)
             print_cmd(cmd)
-            sp.call(cmd, shell=True)
+            sp.call(shlex.split(cmd))
+            print("{} {} is down now".format(name, uuid))
+        except Exception as e:
+            print("Can't down {}: {}".format(uuid, str(e)))
```

Comment:
I don't like that we are catching all the exceptions here. It could hide some other issues. Cant we have:
```suggestion
        except sp.CalledProcessError as e:
            print("Can't down {}: {}".format(uuid, str(e)))
```